cluster-autoscaler/expander/nodecountbalance/nodecountbalance.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodecountbalance
+
+import (
+	"k8s.io/autoscaler/cluster-autoscaler/expander"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+)
+
+type nodecountbalance struct {
+}
+
+// NewStrategy returns an expansion strategy that randomly picks between node groups
+func NewStrategy() expander.Strategy {
+	return &nodecountbalance{}
+}
+
+// BestOption selects from the expansion options to balance node groups in their node counts
+func (r *nodecountbalance) BestOption(expansionOptions []expander.Option, nodeInfo map[string]*schedulercache.NodeInfo) *expander.Option {
+	pos := 0
+	for i, o := range expansionOptions {
+		if i == 0 {
+			continue
+		}
+		if o.NodeCount < expansionOptions[pos].NodeCount {
```

Comment:
@MaciekPytel Thanks for the detailed explanation!
Yours is more general, correct and sophisticated than mine. I like it 👍 
BTW how are you going to determine which node group is equivalent to an another one?
By explicitly passing flags like `--node-group-family=ng1-zone1,ng1-zone2 --node-group-family=ng2-zone1,ng2-zone3`, or maybe in AWS tagging autoscaling groups with family name?